### PR TITLE
Bxmsdoc 2110 Engine installation and introductory information.

### PR DESCRIPTION
**Pull request by gsheldon**
Needs SME review.

### CODE REVIEW

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+[id='optimizer-download-proc']
+= Download {PLANNER}
+
+{PLANNER} is production ready.
+The API is stable but backward incompatible changes can occur.
+
+.Prerequisite
+The {PRODUCT} has been downloaded and installed from the https://access.redhat.com[Red Hat Customer Portal]. See Installing Decision Manager for more information about installing {PRODUCT}.
```
Comment (vidya-iyengar) [inline docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc:8]: Would you be giving the exact location  for downloads later?

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
+[id='optimizer-NP-complete-vs-NP-hard-con']
+= A planning problem is NP-complete or NP-hard
+
+The provided use cases are _probably_ http://en.wikipedia.org/wiki/NP-complete[NP-complete] or harder.
+NP-complete means:
```
Comment (ge0ffrey) [inline docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc:5]: The definition of NP-complete and NP-hard are a pain, because as long as "P vs NP" (see wikipedia) isn't settled, it's hard to phrase it. Anyway, that being said, I would rephrase this as the sentence below so we don't upset the mathematicians:

The provided use cases are _probably_ https://en.wikipedia.org/wiki/NP-completeness[NP-complete/NP-hard],
which means:

Comment (ge0ffrey) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc:5]: I just fixed it in upstream docs too:
https://github.com/kiegroup/optaplanner/commit/1c598641d

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+//@doc-link:Add link to Installation assembly above. 
+
+.Procedure
+. Download the {PLANNER} file from the https://access.redhat.com[Red Hat Customer Portal].
```
Comment (ge0ffrey) [inline docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc:12]: The prerequisite says it needs to be installed alread, but this procedure says that it will be downloaded. Might the prereq be obsolete?

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-hard-vs-soft-constraints-con.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
+[id='optimizer-hard-vs-soft-constraints-con']
+= A planning problem has constraints
```
Comment (ge0ffrey) [inline docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-hard-vs-soft-constraints-con.adoc:2]: I presume, once optaplanner-docs is moved into kie-docs, these sections wouldn't need to be duplicated any more?

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-maven-configuration-proc.adoc [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+. Add a dependency to `optaplanner-core` in your project's `pom.xml` :
++
+
+[source,xml]
+----
+
+  <dependency>
+      <groupId>org.optaplanner</groupId>
+      <artifactId>optaplanner-core</artifactId>
+      <version>...</version>
```
Comment (ge0ffrey) [inline docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-maven-configuration-proc.adoc:20]: {PRODUCT_VERSION}?

**Review by ge0ffrey — APPROVED** [review]

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc (hunk 1 above)
Comment (gsheldon) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc:8]: Included the Enterprise URL attribute.

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc (hunk 2 above)
Comment (gsheldon) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-NP-complete-vs-NP-hard-con.adoc:5]: I've made this: 

"The provided use cases are _probably_ http://en.wikipedia.org/wiki/NP-complete[NP-complete or NP-hard], which means the following statements apply:" 

purely to conform with IBM style guide restrictions on introducing lists with incomplete sentences.

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc (hunk 3 above)
Comment (gsheldon) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-download-proc.adoc:12]: yep, removed.

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-hard-vs-soft-constraints-con.adoc (hunk 4 above)
Comment (gsheldon) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-hard-vs-soft-constraints-con.adoc:2]: that's right it will be a lot easier to just maintain the same changes in the same place, and we need to use this particular anchor type for topic reuse and to build properly.

File: docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-maven-configuration-proc.adoc (hunk 5 above)
Comment (gsheldon) [reply docs/product-business-resource-planner-guide/src/main/asciidoc/optimizer-maven-configuration-proc.adoc:20]: {MAVEN_ARTIFACT_VERSION}.

**Comment by gsheldon** [comment]
These changes have been incorporated and merged in https://github.com/gsheldon/kie-docs/pull/10/.
